crates/bevy_asset/src/path.rs

```diff
@@ -465,6 +465,38 @@ impl<'a> AssetPath<'a> {
         Some(extension)
     }
 
+    /// Returns the path of the [`AssetPath`] relative to the app root.
+    /// Returns None if the [`AssetPath`] uses a non-default Asset Source.
+    ///
+    /// Ex: an [`AssetPath`] referencing "/models/apple.gltf" would return a [`PathBuf`] for "assets/models/apple.gltf"
+    pub fn get_root_relative_path(&self) -> Option<PathBuf> {
+        if let AssetSourceId::Name(_) = self.source {
+            return None;
+        }
+        let path = self.path();
+        let path = if path.is_absolute() {
+            match path.strip_prefix("/") {
+                Ok(path) => path,
+                Err(_) => {
+                    return None;
+                }
+            }
+        } else {
+            path
+        };
+        Some(Path::new(crate::AssetPlugin::DEFAULT_UNPROCESSED_FILE_PATH).join(path))
```

Comment:
Something's bugging me about using the default here, because eventually I think we might end up needing to get the set value of `AssetPlugin::file_path` to cater to folks who don't want to use the default (see #10455 for example). I don't think it blocks merging this though.